src/tests/yamlCompletion.test.ts

```diff
@@ -289,45 +316,138 @@ async function testCompletions(options: TestCompletionOptions) {
     })
 }
 
+// Validate completions that include a full property (key-val pair)
+async function testPropertyCompletions(options: TestPropertyCompletionOptions) {
+    const { labels, json, position  } = options
+
+    const res = await getCompletions(json, position)
+
+    assert.strictEqual(res?.items.length, labels.length)
+
+    const itemLabels = res?.items.map(item => item.label)
+    assert.deepEqual(
+        itemLabels,
+        labels
+    )
+
+    // Test property keys match labels.
+    const itemInsertTextKeys = res?.items.map(item => item.insertText?.split(':')[0])
```

Comment:
We are testing the keys only but I think if we do testing for those we should check the whole thing, not just the key.